mypy/subtypes.py

```diff
@@ -302,6 +331,237 @@ def visit_type_type(self, left: TypeType) -> bool:
+                return False
+            subflags = get_member_flags(member, left.type)
+            superflags = get_member_flags(member, right.type)
+            if IS_SETTABLE in superflags:
+                # Check opposite direction for settable attributes.
+                if not is_subtype(supertype, subtype):
+                    return False
+            if (IS_CLASSVAR in subflags and IS_CLASSVAR not in superflags or
+                    IS_CLASSVAR in superflags and IS_CLASSVAR not in subflags):
+                return False
+            if IS_SETTABLE in superflags and IS_SETTABLE not in subflags:
+                return False
+            # this rule is copied from nominal check in checker.py
+            if IS_CLASS_OR_STATIC in superflags and IS_CLASS_OR_STATIC not in subflags:
+                return False
+    if allow_any:
+        right.type.cache.add((left, right))
+    else:
+        right.type.cache_proper.add((left, right))
+    return True
+
+
+def find_member(name: str, itype: Instance, subtype: Type) -> Optional[Type]:
+    """Find the type of member by 'name' in 'itype's TypeInfo. Apply type arguments
+    from 'itype', and bind 'self' to 'subtype'. Return None if member was not found.
+    """
+    info = itype.type
+    method = info.get_method(name)
+    if method:
+        if method.is_property:
+            assert isinstance(method, OverloadedFuncDef)
+            dec = method.items[0]
+            assert isinstance(dec, Decorator)
+            return find_node_type(dec.var, itype, subtype)
+        return find_node_type(method, itype, subtype)
+    else:
+        # don't have such method, maybe variable or decorator?
+        node = info.get(name)
+        if not node:
+            v = None
+        else:
+            v = node.node
+        if isinstance(v, Decorator):
+            v = v.var
+        if isinstance(v, Var):
+            return find_node_type(v, itype, subtype)
+        if not v and name not in ['__getattr__', '__setattr__', '__getattribute__']:
+            for method_name in ('__getattribute__', '__getattr__'):
+                # Normally, mypy assumes that instances that define __getattr__ have all
+                # attributes with the corresponding return type. If this will produce
+                # many false negatives, then this could be prohibited for
+                # structural subtyping.
+                method = info.get_method(method_name)
+                if method and method.info.fullname() != 'builtins.object':
+                    getattr_type = find_node_type(method, itype, subtype)
+                    if isinstance(getattr_type, CallableType):
+                        return getattr_type.ret_type
+        if itype.type.fallback_to_any:
+            return AnyType(TypeOfAny.special_form)
+    return None
+
+
+def get_member_flags(name: str, info: TypeInfo) -> Set[int]:
+    """Detect whether a member 'name' is settable, whether it is an
```

Comment:
Document the names of all flags that can be included in the set. Again, use a single first line followed by an empty line.